modules/systemd-journal/journal-reader.c

```diff
@@ -598,7 +598,7 @@ _init(LogPipe *s)
   if (!log_source_init(s))
     return FALSE;
 
-  gint res = journald_open(self->journal, SD_JOURNAL_LOCAL_ONLY);
+  gint res = journald_open(self->journal, SD_JOURNAL_LOCAL_ONLY | SD_JOURNAL_ALL_NAMESPACES);
```

Comment:
Resolving the case when the `SD_JOURNAL_ALL_NAMESPACES` does not exists depends on how you resolve the my first comment.
But one way you could do is similar to pcre jit flag:
https://github.com/syslog-ng/syslog-ng/blob/ffe25103987009b7afddce1220ebd6d11820b3f2/lib/compat/pcre.h#L35-L37

Reply:
I'm afraid it's not that easy. `SD_JOURNAL_ALL_NAMESPACES` is defined in an enum in `<systemd/sd-journal.h>`, so the preprocessor won't know if it's defined or not. For example:

```
#include <stdio.h>
#include <systemd/sd-journal.h>

int main() {

printf("%d\n", SD_JOURNAL_ALL_NAMESPACES);

#ifndef SD_JOURNAL_ALL_NAMESPACES
# define SD_JOURNAL_ALL_NAMESPACES 64
#endif

printf("%d\n", SD_JOURNAL_ALL_NAMESPACES);

return 0;

}
```
outputs
```
32
64
```
but i'd expect
```
32
32
```